### PR TITLE
add query v2 support

### PR DESCRIPTION
**Pull request by gangtao**
add query v2/sse support

### CODE REVIEW

File: timeplus/client.go [hunk 1]
```diff
@@ -293,3 +314,110 @@ func (s *TimeplusClient) QueryStream(sql string) (rxgo.Observable, *QueryInfo, e
...
+
+func (s *TimeplusClient) QueryV2Stream(sql string) (rxgo.Observable, error) {
+	query := Query{
+		SQL:         sql,
+		Name:        "",
+		Description: "",
+		Tags:        []string{},
+	}
+
+	createQueryUrl := fmt.Sprintf("%s/queries", s.baseUrlV2())
```
Comment (ye11ow) [inline timeplus/client.go:326]: I use https://github.com/r3labs/sse in our functional test. However, That library doesn't support "POST" version of SSE so I manually copied and pasted the code into our own repo: https://github.com/timeplus-io/neutron/blob/develop/tests/functional/functional_suite_test.go#L1227

I think we can contribute the POST version to them. It should be a minor change

File: timeplus/client.go [hunk 2]
```diff
@@ -15,6 +16,18 @@ import (
 
 const TimeFormat = "2006-01-02 15:04:05.000"
 const APIVersion = "v1beta1"
+const APIV2Version = "v1beta2"
+
+type QueryEvent map[string]any
+type MetricsEvent map[string]any
+type DataEvent [][]any
```
Comment (ye11ow) [inline timeplus/client.go:23]: I think we should create an alias for our `Data` even though it is just `[]any`. In Neutron there are just too many `[]any` so it is hard to tell which one represent the data itself

File: timeplus/client.go (hunk 1 above)
Comment (gangtao) [reply timeplus/client.go:326]: Yes, I did check that lib, we could consider use that lib in the future when we want to publish SDK.

File: timeplus/client.go (hunk 2 above)
Comment (gangtao) [reply timeplus/client.go:23]: Yes, for a complete SDK, like we discussed, we need do type mapping for each cell in the row.  this is more for internal usage for now, let's plan it when we do need a SDK for this.

**Comment by jovezhong** [comment]
(Jove Github Bot) added it to the current sprint.

**Comment by jovezhong** [comment]
(Jove Github Bot) moved this ticket out of the GitHub project(up to 1200 tickets for one project).

**Pull request merged**